### PR TITLE
Working shop

### PR DESCRIPTION
**Pull request by AntonAsbrink**

### CODE REVIEW

File: StortSpelProjekt/Game/src/main.cpp [hunk 1]
```diff
@@ -18,6 +20,7 @@ EnemyFactory enemyFactory;
 int WINAPI WinMain(HINSTANCE hInstance, HINSTANCE, LPSTR, int nCmdShow)
 {
     _CrtSetDbgFlag(_CRTDBG_ALLOC_MEM_DF | _CRTDBG_LEAK_CHECK_DF);
+    //_CrtSetBreakAlloc(82031);
```
Comment (Schtekt) [inline StortSpelProjekt/Game/src/main.cpp]: Unneccessary comment?

File: StortSpelProjekt/Game/src/Gamefiles/Shop.h [hunk 2]
```diff
@@ -40,9 +42,14 @@ class Shop
 	int GetPrice(std::string name);
 	// Get player currency balance
 	int GetPlayerBalance();
+	// Get upgrade image
+	Texture* GetUpgradeImage(std::string name);
```
Comment (Schtekt) [inline StortSpelProjekt/Game/src/Gamefiles/Shop.h]: Why send a whole string object? wouldn't a constant string reference be good enough?

File: StortSpelProjekt/Game/src/Gamefiles/Components/UpgradeComponents/Upgrades/Upgrade.h [hunk 3]
```diff
@@ -24,6 +24,8 @@ class Upgrade
 	// setters and getters for private member variables
 	void SetName(std::string name);
 	std::string GetName() const;
+	void SetImage(std::string imageName);
```
Comment (Schtekt) [inline StortSpelProjekt/Game/src/Gamefiles/Components/UpgradeComponents/Upgrades/Upgrade.h]: Maybe use a const String reference instead?

File: StortSpelProjekt/Game/src/Gamefiles/Components/UpgradeComponents/Upgrades/Upgrade.h [hunk 4]
```diff
@@ -24,6 +24,8 @@ class Upgrade
...
+	std::string GetImage();
```
Comment (Schtekt) [inline StortSpelProjekt/Game/src/Gamefiles/Components/UpgradeComponents/Upgrades/Upgrade.h]: Do you make changes to the string that is given? please add post const keyword.

**Review by Schtekt — DISMISSED** [review]
Please make use of the const keyword otherwise fine.

**Comment by kattalainen** [comment]
Upgrades range velocity and range life steal don't seem to be in the pool of selections

**Review by kattalainen — APPROVED** [review]

**Review by LeoWikstrom — APPROVED** [review]

**Pull request merged**